examples/04_New Models.ipynb

```diff
@@ -835,6 +835,190 @@
     "## Matrix Models"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At this point in the notebook, we have seen a few different model types. In all of these models, we have used dictionaries to represent the states. For example, in the implementation of `ThrownObject_ST` above, see how `dx` is defined with a StateContainer dictionary. While all models can be constructed in this way, Matrix Models provide an advanced feature that is applicable and useful for some use-cases. \n",
+    "\n",
+    "Matrix models are a particular type of model that represent the state of the system using matrices instead of dictionaries. The provided model.StateContainer, InputContainer, and OutputContainer can be treated as dictionaries but use an underlying matrix. This is important for some applications like surrogate and machine-learned models where the state is represented by a tensor. In a matrix model, simulation functions propagate the state using the matrix form, preventing the inefficiency of having to convert to and from dictionaries."
```

Comment:
It's also faster than recreating the StateContainer each time, especially when you update inplace.